nni/algorithms/compression/v2/pytorch/base/scheduler.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List, Dict, Tuple
+
+from torch.nn import Module
+from torch.tensor import Tensor
+
+
+class BasePruningScheduler:
+    def generate_task(self) -> Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]]]:
+        """
+        Returns
+        -------
+        Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]]]
+            Return new task id, model under pruning, config list used in this task and pre-masks.
+        """
+        raise NotImplementedError()
+
+    def record_task_result(self, task_id: int, pruned_model: Module, masks: Dict[str, Dict[str, Tensor]], score: float,
+                           up_model_masks: Dict[str, Dict[str, Tensor]]):
+        """
+        Used to record the task result.
+
+        Parameters
+        ----------
+        task_id
+            The id of the finished task.
+        pruned_model
+            The pruned model after `pruning_one_step`.
+        masks
+            The masks should be applied on the pruned model.
+        score
+            The score of the pruning performance in this task.
+        up_model_masks
+            The masks applied on the under pruning model.
+        """
+        raise NotImplementedError()
+
+    def pruning_one_step(self, model: Module, config_list: List[Dict], masks: Dict[str, Dict[str, Tensor]]) \
+            -> Tuple[Module, Dict[str, Dict[str, Tensor]], float, Dict[str, Dict[str, Tensor]]]:
+        """
+        Pruning the model with config list.
+
+        Parameters
+        ----------
+        model
+            The model under pruning.
```

Comment:
what is the meaning of "under pruning"?

Reply:
maybe we can say, it is a normal pytorch module. no wrapper, no mask in `model` right?

Reply:
Yes, this model is an unwrapped model. "under pruning" means this model is the model before pruning and will be pruned in this step. will update the docstring.

Reply:
fix it